zeppelin-server/src/main/java/org/apache/zeppelin/service/NotebookService.java

```diff
@@ -955,37 +952,46 @@ public void emptyTrash(ServiceContext context,
   }
 
   public void spell(String noteId,
-                    Message message,
+                    String paragraphId,
+                    String title,
+                    String text,
+                    Object results,
```

Comment:
InterpreterResult?

Reply:
yes, it's InterpreterResult, I think we should do as much as possible logic inside the service class, not in a facade.